hibernate-reactive-core/src/main/java/org/hibernate/reactive/session/impl/ReactiveStatelessSessionImpl.java

```diff
@@ -223,23 +223,24 @@ public ReactiveEntityPersister getEntityPersister(String entityName, Object obje
     public CompletionStage<Void> reactiveUpdate(Object entity) {
         checkOpen();
         if ( entity instanceof HibernateProxy ) {
+            Object proxy = entity;
             final LazyInitializer hibernateLazyInitializer = ( (HibernateProxy) entity ).getHibernateLazyInitializer();
             if ( hibernateLazyInitializer.isUninitialized() ) {
-                return reactiveFetch( entity, true ).thenCompose(
-                        fetchedEntity -> executeReactiveUpdate( fetchedEntity ) );
+                return reactiveFetch( entity, true )
+                        .thenCompose( fetchedEntity -> executeReactiveUpdate(proxy, fetchedEntity) );
             }
             else {
-                return executeReactiveUpdate( hibernateLazyInitializer.getImplementation() );
+                return executeReactiveUpdate( proxy, hibernateLazyInitializer.getImplementation() );
             }
         }
 
-        return executeReactiveUpdate( entity );
+        return executeReactiveUpdate( null,  entity );
     }
 
-    private CompletionStage<Void> executeReactiveUpdate(Object entity) {
+    private CompletionStage<Void> executeReactiveUpdate(Object proxy, Object entity) {
         ReactiveEntityPersister persister = getEntityPersister( null, entity );
         Serializable id = persister.getIdentifier( entity, this );
-        Object[] state = persister.getPropertyValues( entity );
+        Object[] state = proxy != null ? persister.getPropertyValues( proxy ) : persister.getPropertyValues( entity );
```

Comment:
I know that I said that passing the same value `entity` for both values was weird. It turns out I was wrong and it would make sense in this case. There  is really no need to add an if-then here. But, please, add a comment to explain that we expect proxy to contains the state that we want to save in the database

Reply:
So yes, I can remove the if-then and change the non-proxy call to:  `executeReactiveUpdate( entity,  entity );`

For me, it's a little confusing just looking in the `executeReactiveUpdate(proxy, entity)` method because it makes it appear to always be a proxy object... but I can do that.

Reply:
I think renaming a parameter is easier than to add extra logic, but that's me :-)